### PR TITLE
Feature/sc 190

### PR DESCRIPTION
**Pull request by CaptainOfHacks**
- In EU action timeline derive categories feature from the names of spokepersons 

- extend eu timeline index to include topics that is an array of strings

### CODE REVIEW

File: sem_covid/__init__.py [hunk 1]
```diff
@@ -198,6 +198,18 @@ def ELASTICSEARCH_USERNAME(self) -> str:
     def ELASTICSEARCH_PASSWORD(self) -> str:
         return BaseConfig.find_value()
 
+<<<<<<< Updated upstream
```
Comment (costezki) [inline sem_covid/__init__.py:201]: let's resolve the merge

File: .gitignore [hunk 2]
```diff
@@ -160,3 +160,7 @@ junit_report.xml
 *.bbl
 *.blg
 /report.html
+tests/ml_experiments/webserver_config.py
```
Comment (costezki) [inline .gitignore:163]: I would simply add tests/ml_experiments folder all together

File: sem_covid/services/crawlers/scrapy_crawlers/spiders/eu_timeline_spider.py [hunk 3]
```diff
@@ -61,6 +63,12 @@ def parse_main_page(self, response):
                     else:
                         self.data.append(meta)
 
+    def _get_topics_by_spoke_person_name(self, spoke_person_name: str) -> list:
```
Comment (costezki) [inline sem_covid/services/crawlers/scrapy_crawlers/spiders/eu_timeline_spider.py:66]: fair enough

File: sem_covid/services/crawlers/scrapy_crawlers/spiders/eu_timeline_spider.py [hunk 4]
```diff
@@ -92,10 +100,13 @@ def parse_presscorner_page(self, response):
             'href')
 
         item['press_contacts'] = list()
+        item['topics'] = list()
         press_contacts = response.xpath('//ul[@class="ecl-listing"]/li')
         for press_contact in press_contacts:
+            document_spoke_person_name = press_contact.xpath('*//div/h4/text()').get()
+            item['topics'].append(self.get_topics_by_spoke_person_name(document_spoke_person_name))
```
Comment (costezki) [inline sem_covid/services/crawlers/scrapy_crawlers/spiders/eu_timeline_spider.py:107]: if you append, you will have an array inside another array; simply assign the value instead

File: tests/unit/test_crawl_eu_action.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    assert test_path
+    assert 'timeline_press' in test_path
+
+
+def test_spoke_person_content():
+    df_spoke_person = pd.read_json(config.CRAWLER_EU_TIMELINE_SPOKEPERSONS)
+    assert not df_spoke_person.empty
+    assert 'Name' in df_spoke_person.columns
+    assert 'Johannes Bahrke' in df_spoke_person['Name'].values
+    assert df_spoke_person[df_spoke_person['Name'] == 'Johannes Bahrke']['Topics'][0]
```
Comment (costezki) [inline tests/unit/test_crawl_eu_action.py:24]: I would add also a negative test; if the person does not exist, the result shall be an empty array

**Review by costezki — APPROVED** [review]
fix the array in array possible result (see comment); and then we can merge; 
the implementation looks good
